tools/c7n_azure/tests/test_keyvault-cert.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2015-2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from azure_common import BaseTest, arm_template
+
+
+class KeyVaultCertTest(BaseTest):
+    def setUp(self):
+        super(KeyVaultCertTest, self).setUp()
+
+    @arm_template('keyvault-cert.json')
+    def test_find_by_name(self):
```

Comment:
If you compare the cert age, you might need to mock datetime to return some fixed date.. Otherwise your condition lt 4 can fail based on the run date.